Game/space-invaders/src/enemies/EnemyManager.h

```diff
@@ -79,4 +81,12 @@ class EnemyManager
     void SpawnEnemies(const LevelDefinition& level);
     glm::vec2 CalculateEnemySize(const LevelDefinition& level);
     glm::vec2 CalculateStartPosition(const LevelDefinition& level);
+
+    struct Emmiter
```

Comment:
Stead of using this struct to keep track of what emmiter to use, I belive it will be better to just having a `int lastUsedEmmiter` to make a circular buffer. The idea is every time we use a emmiter to spawn the death particles, we increment the index, respecting the bounds and going back to zero when used all emmiter (`index = (index + 1) % totalEmmiters`). This way we would decrease the chances of using an emmiter which is still emmiting. If using a still in use emmiter happens frequently, this means we need to increase our emmiter pool.